src/pkg/services/m365/api/lists_test.go

```diff
@@ -68,6 +69,389 @@ func (suite *ListsUnitSuite) TestBytesToListable() {
+
+			require.NotEmpty(t, newCd)
+
+			if test.checkNil {
+				assert.Nil(t, newCd.GetValidation())
+			} else {
+				assert.NotNil(t, newCd.GetValidation())
+			}
+		})
+	}
+}
+
+func (suite *ListsUnitSuite) TestColumnDefaultValueSetWhenProvided() {
+	tests := []struct {
+		name     string
+		getOrig  func() models.ColumnDefinitionable
+		checkNil bool
+	}{
+		{
+			name: "column default value not set",
+			getOrig: func() models.ColumnDefinitionable {
+				textColumn := models.NewTextColumn()
+
+				cd := models.NewColumnDefinition()
+				cd.SetText(textColumn)
+
+				return cd
+			},
+			checkNil: true,
+		},
+		{
+			name: "column default value set",
+			getOrig: func() models.ColumnDefinitionable {
+				defaultVal := "some-val"
+
+				textColumn := models.NewTextColumn()
+
+				colDefaultVal := models.NewDefaultColumnValue()
+				colDefaultVal.SetValue(ptr.To(defaultVal))
+
+				cd := models.NewColumnDefinition()
+				cd.SetText(textColumn)
+				cd.SetDefaultValue(colDefaultVal)
+
+				return cd
+			},
+		},
+	}
+
+	for _, test := range tests {
+		suite.Run(test.name, func() {
+			t := suite.T()
+
+			orig := test.getOrig()
+			newCd := cloneColumnDefinitionable(orig)
+
+			require.NotEmpty(t, newCd)
+
+			if test.checkNil {
+				assert.Nil(t, newCd.GetDefaultValue())
+			} else {
+				assert.NotNil(t, newCd.GetDefaultValue())
+				assert.Equal(t, "some-val", ptr.Val(newCd.GetDefaultValue().GetValue()))
+			}
+		})
+	}
+}
+
+func (suite *ListsUnitSuite) TestCheckColumnType() {
+	tests := []struct {
+		name    string
+		getOrig func() models.ColumnDefinitionable
+		checkFn func(models.ColumnDefinitionable) bool
+	}{
+		{
+			name: "column type should be number",
+			getOrig: func() models.ColumnDefinitionable {
+				numColumn := models.NewNumberColumn()
+
+				cd := models.NewColumnDefinition()
+				cd.SetNumber(numColumn)
+
+				return cd
+			},
+			checkFn: func(cd models.ColumnDefinitionable) bool {
+				return cd.GetNumber() != nil
+			},
+		},
+		{
+			name: "column type should be person or group",
+			getOrig: func() models.ColumnDefinitionable {
+				pgColumn := models.NewPersonOrGroupColumn()
+
+				cd := models.NewColumnDefinition()
+				cd.SetPersonOrGroup(pgColumn)
+
+				return cd
+			},
+			checkFn: func(cd models.ColumnDefinitionable) bool {
+				return cd.GetPersonOrGroup() != nil
+			},
+		},
+		{
+			name: "column type should default to text",
+			getOrig: func() models.ColumnDefinitionable {
+				return models.NewColumnDefinition()
+			},
+			checkFn: func(cd models.ColumnDefinitionable) bool {
+				return cd.GetText() != nil
+			},
+		},
+	}
+
+	for _, test := range tests {
+		suite.Run(test.name, func() {
+			t := suite.T()
+
+			orig := test.getOrig()
+			newCd := cloneColumnDefinitionable(orig)
+
+			require.NotEmpty(t, newCd)
+			assert.True(t, test.checkFn(newCd))
+		})
+	}
+}
+
+func (suite *ListsUnitSuite) TestLegacyColumnsAreNotSet() {
+	listName := "test-list"
+	contentTypeColumnName := "ContentType"
```

Comment:
suggestion: turn these values into constants.  Use those consts as both the keys in your list structure maps, as well as here in the tests.